sos/report/plugins/dellrac.py

```diff
@@ -0,0 +1,63 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
+import re
+import time
+
+
+class DellRAC(Plugin, IndependentPlugin):
+
+    short_desc = 'Dell Remote Access Controller Administration'
+
+    plugin_name = 'dellrac'
+    profiles = ('system', 'storage', 'hardware',)
+    packages = ('srvadmin-idracadm7',)
+
+    option_list = [
+        PluginOpt('debug', default=False, desc='capture support assist data')
+    ]
+
+    racadm = '/opt/dell/srvadmin/bin/idracadm7'
+
+    def setup(self):
+        for subcmd in ['getniccfg', 'getsysinfo']:
+            self.add_cmd_output(
+                '%s %s' % (self.racadm, subcmd),
+                suggest_filename='racadm_%s' % subcmd)
+
+        if self.get_option("debug"):
+            self.do_debug()
+
+    def do_debug(self):
+        self.add_cmd_output(
+            '%s supportassist accepteula',
+            suggest_filename='racadm_supportassist_accepteula')
+
+        logpath = self.get_cmd_output_path()
+        status = self.collect_cmd_output(
+            '%s supportassist collect' % self.racadm,
+            suggest_filename='racadm_supportassist_collect')
+
+        jid = ''
+        if status['status'] == 0:
+            jid = re.search(r'JID_[0-9]*', status['output']).group()
+
+        done_collect = False
+        while not done_collect and jid:
+            time.sleep(10)
+            status = self.collect_cmd_output(
+                '%s jobqueue view -i %s' % (self.racadm, jid),
+                suggest_filename='racadm_jobqueue_view')
+            done_collect = re.search(r'Status=Completed', status['output'])
```

Comment:
What are we waiting for to be completed here? Does the previous command automatically background itself? Is there someway to not return until that command is collected? I'd prefer that method as that allows sos to leverage its timeout controls.

Reply:
Yes, the previous command goes in background. There is no option to run that in foreground. The only way is to keep checking job id status.